components/Handlers.js

```diff
@@ -15,28 +15,81 @@ func ${operation}(msg *message.Message) error {
 }
 `;
 
-function SubscriptionHandlers({ channels }) {
+export function SubscriptionHandlers({ channels }) {
   return Object.entries(channels)
     .map(([channelName, channel]) => {
       if (channel.hasPublish()) {
         const operation = pascalCase(channel.publish().id());
```

Comment:
Exactly! Either:
- some method (handler function) name is auto generated (and have it documented)
- or the docs should explicit that `operationId` should be used to properly define your handler name

Reply:
And there is also `$id` property of `payload`, based on which the proper Go `struct` naming is done.
That's another useful and optional thing (without it you get a name, but it's something like `AnonymousSchema` which is not that not that great. It's ok that it is generated, no syntax errors, but the naming can be improved).

Reply:
@smoya @dxps thanks for noticing this.
I am tempted to generate a method name based if `operationId` is not present.
But as @dxps mentions we would have to make sure there are no duplicate names.
Another thing I notice is that channel names can have special characters. So we would have to remove all special characters.
I like @dxps's suggestion to mandate an `operationId` for this spec. Meanwhile I can also check other generators on how they handle this scenario

Reply:
@anandsunderraman I reviewed the code again and I noticed you didn't add a a guard checking if the `operationId` is present, forcing the user to set it or throwing an error otherwise. You have an example in [nodejs-ws-template](https://github.com/asyncapi/nodejs-ws-template/blob/c50a34556de766098e7b3b25d95fc064235e3728/filters/all.js#L21-L28).

Also, I would mention this in the README file as for example [java-spring-template](https://github.com/asyncapi/java-spring-template/blob/7feab3c2d5b2800828e25e28c30eb0deed56e90d/README.md#asyncapi-definitions) does.

Reply:
Good catch! Forgot about `operationId`, although I mentioned it initially.

The other part that could get improved is related to the `$id` property of payload, based on which the proper Go struct naming is done. That's another useful and optional thing, without specifying it you get a name like `AnonymousSchema` which is not that great or useful.